dataclients/kubernetes/clusterclient.go

```diff
@@ -177,6 +192,33 @@ func newClusterClient(o Options, apiURL, ingCls, rgCls string, quit <-chan struc
 	return c, nil
 }
 
+// serializes a given map of label selectors to a string that can be appended to a request URI to K8S
+// Examples (note that the resulting value in the query is URL escaped, for readability thi is not done in examples):
+// 	[] becomes ``
+// 	["label": ""] becomes `?labelSelector=label`
+// 	["label": "value"] becomes `?labelSelector=label=value`
+// 	["label": "value", "label2": "value2"] becomes `?labelSelector=label=value&label2=value2`
+func toLabelSelectorQuery(selectors map[string]string, defaultSelectors map[string]string) string {
+	if selectors == nil {
+		selectors = defaultSelectors
+	}
+
+	if len(selectors) == 0 {
+		return ""
+	}
+
+	var strs []string
+	for k, v := range selectors {
+		if v == "" {
+			strs = append(strs, k)
+		} else {
+			strs = append(strs, fmt.Sprintf(labelSelectorFmt, k, v))
+		}
+	}
+
+	return fmt.Sprintf(labelSelectorQueryFmt, url.QueryEscape(strings.Join(strs, ",")))
```

Comment:
Since we already have `map[string]string` wouldn't it make more sense to just make an additional method something like:
`getJSONWithQueryParameter` and pass along the corresponding query params?

This would also make it as such that we don't need to gather everything here and query escape the values.

Reply:
This would essentially mean that it would be built to a string with each request even though it never changes with the given setup, so can be just done once and appended to the URI.